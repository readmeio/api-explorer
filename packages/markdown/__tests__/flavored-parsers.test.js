const unified = require('unified');
const remarkParse = require('remark-parse');
const rehypeSanitize = require('rehype-sanitize');

const parseCallouts = require('../processor/parse/flavored/callout');
const parseCodeTabs = require('../processor/parse/flavored/code-tabs');
const options = require('../processor/options.json').markdownOptions;

const sanitize = { attributes: [], tagNames: [] };
const process = (text, opts = options) =>
  text &&
  unified()
    .use(remarkParse, opts)
    .data('settings', { position: false })
    .use([parseCallouts.sanitize(sanitize), parseCodeTabs.sanitize(sanitize)])
    .use(rehypeSanitize)
    .parse(text);

describe('Parse ReadMe-Flavored Markdown Syntax', () => {
  it('Callouts', () => {
    const text = `> ℹ️ Info Callout
    >
    > Lorem ipsum dolor  sit amet consectetur adipisicing elit.`;
    expect(process(text)).toMatchSnapshot();
  });

  it('Multi-Block', () => {
    const text =
      "\n\n```javascript multiple.js\nconsole.log('a multi-file code block');\n```\n```javascript\nconsole.log('an unnamed sample snippet');\n```\n\n&nbsp;";
    const ast = process(text);
    expect(ast).toMatchSnapshot();
  });

  it('Single Block', () => {
    const text = "\n\n```javascript multiple.js\nconsole.log('a multi-file code block');\n```\n\n&nbsp;";
    const ast = process(text);
    expect(ast).toMatchSnapshot();
  });
});
