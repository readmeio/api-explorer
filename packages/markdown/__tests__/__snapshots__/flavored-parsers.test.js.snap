// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse ReadMe-Flavored Markdown Syntax Callouts 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "ℹ️ Info Callout",
            },
          ],
          "type": "paragraph",
        },
      ],
      "data": Object {
        "hName": "rdme-callout",
        "hProperties": Object {
          "icon": "ℹ️",
          "theme": "info",
          "title": "Info Callout",
          "value": "",
        },
      },
      "type": "rdme-callout",
    },
    Object {
      "lang": null,
      "meta": null,
      "type": "code",
      "value": ">
> Lorem ipsum dolor  sit amet consectetur adipisicing elit.",
    },
  ],
  "type": "root",
}
`;

exports[`Parse ReadMe-Flavored Markdown Syntax Multi-Block 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "className": "tab-panel",
          "data": Object {
            "hName": "code",
            "hProperties": Object {
              "lang": "javascript",
              "meta": "multiple.js",
            },
          },
          "lang": "javascript",
          "meta": "multiple.js",
          "type": "code",
          "value": "console.log('a multi-file code block');",
        },
        Object {
          "className": "tab-panel",
          "data": Object {
            "hName": "code",
            "hProperties": Object {
              "lang": "javascript",
              "meta": null,
            },
          },
          "lang": "javascript",
          "meta": null,
          "type": "code",
          "value": "console.log('an unnamed sample snippet');",
        },
      ],
      "className": "tabs",
      "data": Object {
        "hName": "code-tabs",
      },
      "type": "code-tabs",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": " ",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`Parse ReadMe-Flavored Markdown Syntax Single Block 1`] = `
Object {
  "children": Array [
    Object {
      "className": "tab-panel",
      "data": Object {
        "hName": "code",
        "hProperties": Object {
          "lang": "javascript",
          "meta": "multiple.js",
        },
      },
      "lang": "javascript",
      "meta": "multiple.js",
      "type": "code",
      "value": "console.log('a multi-file code block');",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": " ",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;
